### PR TITLE
Fix "Fertilise when variable reaches threshold" Manager

### LINKED ISSUE #8977: "Fertilise when variable reaches threshold" of "Management Toolbox" malfunctions
### What happened?

If the target variable has a small magnitude, this manager might be triggered multiple times.

image

image

### What Operating System are you on?

Windows

### PR DESCRIPTION
**Pull request by BrianCollinss**
Resolves #8977

### CODE REVIEW

File: ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx [hunk 1]
```diff
@@ -1397,30 +1397,43 @@
...
+            "\t            if (obj is double)",
+            "\t                value = (double) obj;",
+            "\t            else if (obj is IFunction f)",
+            "\t                value = f.Value();",
+            "\t            else",
+            "\t                throw new Exception($\"Unknown type of variable: {VariableName}\");\t                \t           ",
+            "\t                 ",
+            "\t            if (!wasTriggered && MathUtilities.IsGreaterThanOrEqual(value, TriggerValue)) {",
+            "\t                FertiliseDate = clock.Today.AddDays(NumDaysToWait);",
+            "\t                wasTriggered = true;                ",
```
Comment (hol353) [inline ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx:1430]: The logic of this looks wrong e.g. this if statement will only ever be true once per sowing.

File: ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx [hunk 2]
```diff
@@ -1397,30 +1397,43 @@
...
+            "\t            }",
+            "",
+            "\t            if (!wasApplied && clock.Today == FertiliseDate) {",
+            "\t            \tfertiliser.Apply(Amount: Amount, Type: FertiliserType);\t",
+            "\t            \twasApplied = true;",
+            "\t            }\t                          ",
```
Comment (hol353) [inline ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx:1436]: Should this if statement use wasTriggered. The previous if statement is no longer connected to this if statement.

File: ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx [hunk 3]
```diff
@@ -1513,6 +1526,10 @@
             "}"
           ],
           "Parameters": [
+            {
+              "Key": "EventName",
+              "Value": "[Wheat].Phenology.PlantEmerged"
+            },
```
Comment (hol353) [inline ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx:1532]: Where did this come from? This isn't in the tip revision on origin, nor is it used by the script.

**Review by hol353 — CHANGES_REQUESTED** [review]

**Comment by hol353** [comment]
I think a simpler approach to solve the problem is to put an else statement on the first if statement that has something like:

```
FertiliseDate = DateTime.MinValue;
```

File: ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx (hunk 1 above)
Comment (BrianCollinss) [reply ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx:1430]: Isn't it the intention? The original IF statement would check if the value is within 0.1 of the target assuming that would happen once.

File: ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx (hunk 3 above)
Comment (BrianCollinss) [reply ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx:1532]: No idea.

**Comment by BrianCollinss** [comment]
Happy to leave this to someone more knowledgeable.

**Pull request closed without merging**